plugins/android_boot/fu-android-boot-device.c

```diff
@@ -0,0 +1,441 @@
+/*
+ * Copyright (C) 2022 Dylan Van Assche <me@dylanvanassche.be>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+#include <fcntl.h>
+#include <string.h>
+
+#include "fu-android-boot-device.h"
+
+#define PROC_CMDLINE	 "/proc/cmdline"
```

Comment:
needing to read `/proc/cmdline` probably means this plugin should only run when the OS is linux (which I guess you've implicitly covered by gudev requirement)

Reply:
Probably yes, I didn't add a check here for that, but using the internal cmdline parsing of fwupd might solve this already?

Reply:
Probably yes, I didn't add a check here for that, but using the internal cmdline parsing of fwupd might solve this already?